BootloaderCorePkg/Library/S3SaveRestoreLib/S3SaveRestore.c

```diff
@@ -0,0 +1,131 @@
+/** @file
+
+  Copyright (c) 2017, Intel Corporation. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <PiPei.h>
+#include <Library/DebugLib.h>
+#include <Library/HobLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/SocInitLib.h>
+#include <Library/MpInitLib.h>
+#include <Library/IoLib.h>
+#include <Guid/SmmInformationGuid.h>
+
+
+S3_SAVE_REG   *mS3SaveReg = NULL;
```

Comment:
Can we have better way to avoid global variable in Library?

Reply:
Just wanted to reuse the S3SaveReg offset calculated the first time. But assuming there wont be many registers to be saved for S3 resume, if at all, will just calculate this offset each time we call into this function. We can avoid global here.